app/timetables/models.py

```diff
@@ -296,3 +304,88 @@ def __str__(self):
 
     class Meta:
         unique_together = ('timetable', 'vendor')
+
+
+class ServingAutoUpdate(models.Model):
+    """Model representing Automatic Update of Servings."""
+
+    timetable = models.ForeignKey(Timetable, on_delete=models.CASCADE)
+    vendor = models.ForeignKey(Vendor, on_delete=models.CASCADE)
+    date = models.DateField()
+
+    @staticmethod
+    def get_menu_items(timetable, date):
+        cycle_day = timetable.calculate_cycle_day(date)
+        menu_items = MenuItem.objects.filter(
+            timetable=timetable,
+            cycle_day=cycle_day
+        )
+
+        if not menu_items:
+            raise ValidationError(
+                _('No matching menu_item for this Timetable and Date combination.')
+            )
+
+        return menu_items
+
+    @staticmethod
+    def verify_vendor_is_serving(timetable, vendor, date):
+        if not vendor.is_vendor_serving(timetable, date):
+            raise ValidationError(
+                _('Ensure the specified Vendor has an active tenure ' +
+                    'for the specified Date on the specified Timetable')
+            )
+
+    @classmethod
+    def get_servings(cls, timetable, vendor, date):
+        cls.verify_vendor_is_serving(timetable, vendor, date)
+        menu_items = cls.get_menu_items(timetable, date)
+
+        kwargs = {
+            'timetable': timetable,
+            'vendor': vendor,
+            'date': date
+        }
+        if not cls.objects.filter(**kwargs).exists():
+            cls.objects.create(
+                timetable=timetable,
```

Comment:
How about you just pass in `**kwargs` to `cls.objects.create` here to avoid repetition ?